src/dayview.html

```diff
@@ -1,65 +1,167 @@
+                  <table class="table dayview-allday-content-table">
                     <tbody>
-                        <tr *ngFor="let tm of views[0].rows; let i = index">
-                            <td class="calendar-hour-column text-center">
-                                {{hourColumnLabels[i]}}
-                            </td>
-                            <td class="calendar-cell">
-                                <ng-template [ngTemplateOutlet]="dayviewInactiveNormalEventSectionTemplate"
-                                    [ngTemplateOutletContext]="{tm:tm, hourParts: hourParts}">
-                                </ng-template>
+                      <tr>
+                        <ng-container *ngIf="sliderIndex === currentViewIndex">
+                          <ng-container *ngFor="let allDayEventsCategory of views[sliderIndex].categorizedAllDayEvents">
+                            <td *ngIf="allDayEventsCategory[0]?.event?.category === category"
+                              class="calendar-cell" [ngClass]="{'calendar-event-wrap': allDayEventsCategory.length > 0}"
+                              [ngStyle]="{height: 25 * allDayEventsCategory.length + 'px'}">
+                              <ng-template [ngTemplateOutlet]="dayviewAllDayEventSectionTemplate"
+                                [ngTemplateOutletContext]="{allDayEvents: allDayEventsCategory, eventTemplate: dayviewAllDayEventTemplate}">
+                              </ng-template>
                             </td>
-                        </tr>
+                          </ng-container>
+                        </ng-container>
+                        <td class="calendar-cell" *ngIf="sliderIndex !== currentViewIndex">
+                          <ng-template [ngTemplateOutlet]="dayviewInactiveAllDayEventSectionTemplate"
+                            [ngTemplateOutletContext]="{allDayEvents: views[sliderIndex].allDayEvents}">
+                          </ng-template>
+                        </td>
+                      </tr>
                     </tbody>
-                </table>
-            </init-position-scroll>
+                  </table>
+                </td>
+              </tbody>
+            </table>
+          </div>
         </div>
+        <init-position-scroll *ngIf="sliderIndex === currentViewIndex"
+          class="dayview-normal-event-container dayview-categorized-normal-event-container"
+          [initPosition]="initScrollPosition" [emitEvent]="preserveScrollPosition"
+          (onScroll)="setScrollPosition($event)">
+          <table class="table table-bordered table-fixed dayview-normal-event-table">
+            <tbody>
+              <tr *ngFor="let row of views[sliderIndex].rows; let i = index">
+                <td class="calendar-hour-column text-center">
+                  {{ hourColumnLabels[i] }}
+                </td>
+                <td *ngIf="views[sliderIndex].categories.length === 0" class="calendar-cell" (click)="select(row)"
```

Comment:
is this necessary, just assume if need to show categorized view, always have category number larger than 0

Reply:
yes it's necessary, it's used as a placeholder but only there is no event. Remove it then the hour column 'td' will be the only element which occupies whole row like this:
![Uploading image.png…]()

Reply:
![image](https://github.com/twinssbc/Ionic2-Calendar/assets/27603495/f33986ef-fede-4c8d-ac6f-9d64a64b5710)

Reply:
why not using dayviewCategorySource to generate all td? it is the predefined static list of all category names